test/integration/controller/core/capacitycontroller_test.go

```diff
@@ -0,0 +1,186 @@
+			Resource(testing.MakeResource(resourceGPU).
+				Flavor(testing.MakeFlavor(flavorModelA, "5").Ceiling("10").Obj()).
+				Flavor(testing.MakeFlavor(flavorModelB, "5").Ceiling("10").Obj()).Obj()).Obj()
+		gomega.Expect(k8sClient.Create(ctx, capacity)).To(gomega.Succeed())
+		emptyCapStatus = kueue.CapacityStatus{
+			AssignedWorkloads: 0,
+			UsedResources: kueue.ResourceUsage{
+				corev1.ResourceCPU: {
+					flavorOnDemand: {Total: pointer.Quantity(resource.MustParse("0"))},
+					flavorSpot:     {Total: pointer.Quantity(resource.MustParse("0"))},
+				},
+				resourceGPU: {
+					flavorModelA: {Total: pointer.Quantity(resource.MustParse("0"))},
+					flavorModelB: {Total: pointer.Quantity(resource.MustParse("0"))},
+				},
+			},
+		}
+	})
+
+	ginkgo.AfterEach(func() {
+		if capacity != nil {
+			err := k8sClient.Delete(ctx, capacity)
+			if !errors.IsNotFound(err) {
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			}
+		}
+	})
+
+	ginkgo.It("Should update status when workloads are assigned and finish", func() {
+		workloads := []*kueue.QueuedWorkload{
+			testing.MakeQueuedWorkload("one", ns.Name).
+				Request(corev1.ResourceCPU, "2").AssignFlavor(corev1.ResourceCPU, flavorOnDemand).
+				Request(resourceGPU, "2").AssignFlavor(resourceGPU, flavorModelA).Obj(),
+			testing.MakeQueuedWorkload("two", ns.Name).
+				Request(corev1.ResourceCPU, "3").AssignFlavor(corev1.ResourceCPU, flavorOnDemand).
+				Request(resourceGPU, "3").AssignFlavor(resourceGPU, flavorModelA).Obj(),
+			testing.MakeQueuedWorkload("three", ns.Name).
+				Request(corev1.ResourceCPU, "1").AssignFlavor(corev1.ResourceCPU, flavorOnDemand).
+				Request(resourceGPU, "1").AssignFlavor(resourceGPU, flavorModelB).Obj(),
+			testing.MakeQueuedWorkload("four", ns.Name).
+				Request(corev1.ResourceCPU, "1").AssignFlavor(corev1.ResourceCPU, flavorSpot).
+				Request(resourceGPU, "1").AssignFlavor(resourceGPU, flavorModelB).Obj(),
+			testing.MakeQueuedWorkload("five", ns.Name).
+				Request(corev1.ResourceCPU, "1").AssignFlavor(corev1.ResourceCPU, flavorSpot).
+				Request(resourceGPU, "1").AssignFlavor(resourceGPU, flavorModelB).Obj(),
+			testing.MakeQueuedWorkload("six", ns.Name).
+				Request(corev1.ResourceCPU, "1").AssignFlavor(corev1.ResourceCPU, flavorSpot).
+				Request(resourceGPU, "1").AssignFlavor(resourceGPU, flavorModelA).Obj(),
+		}
+
+		ginkgo.By("Creating workloads")
+		for _, w := range workloads {
+			gomega.Expect(k8sClient.Create(ctx, w)).To(gomega.Succeed())
+		}
+		gomega.Eventually(func() kueue.CapacityStatus {
+			var updatedCap kueue.Capacity
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(capacity), &updatedCap)).To(gomega.Succeed())
+			return updatedCap.Status
+		}, timeout, interval).Should(testing.Equal(emptyCapStatus))
+
+		ginkgo.By("Assigning workloads")
+		assignments := []kueue.CapacityReference{"capacity", "capacity", "capacity", "capacity", "other", ""}
```

Comment:
`capacity.Name` just so it is clear we are assigning to the capacity created earlier

Reply:
Done